tests/integration/raw_studies_blueprint/test_fetch_raw_data.py

```diff
@@ -297,206 +298,405 @@ def test_get_study(self, client: TestClient, user_access_token: str, internal_st
+        user_access_token: str,
+        internal_study_id: str,
+    ):
+        """
+        Test the `get_study_file` endpoint for fetching for a file in its original format.
+
+        This test retrieves a specific file from a study identified by a UUID and checks
+
+        The test performs the following steps:
+        1. Copies the user resources in the Study directory.
+        2. Uses the API to download a file from the "user/folder" directory.
+        3. Compares the fetched data with the expected file from disk.
+        4. Check for cases where Errors should be returned.
+        """
+        # First copy the user resources in the Study directory
+        with db():
+            study: RawStudy = db.session.get(Study, internal_study_id)
+            study_dir = pathlib.Path(study.path)
+        client.headers = {"Authorization": f"Bearer {user_access_token}"}
+        original_file_url = f"/v1/studies/{internal_study_id}/raw/original-file"
+
+        shutil.copytree(
+            ASSETS_DIR.joinpath("user"),
+            study_dir.joinpath("user"),
+            dirs_exist_ok=True,
+        )
+
+        # Then, use the API to download the files from the "user/folder" directory
+        user_folder_dir = study_dir.joinpath("user/folder")
+        for file_path in user_folder_dir.glob("*.*"):
+            rel_path = file_path.relative_to(study_dir).as_posix()
+            res = client.get(original_file_url, params={"path": rel_path})
+            assert res.status_code == 200, res.json()
+            actual = res.content
+            expected = file_path.read_bytes()
+            assert actual == expected
+
+        # If the extension is unknown, we should have a "binary" content
+        user_folder_dir = study_dir.joinpath("user/unknown")
+        for file_path in user_folder_dir.glob("*.*"):
+            rel_path = file_path.relative_to(study_dir)
+            res = client.get(original_file_url, params={"path": f"/{rel_path.as_posix()}"})
+            assert res.status_code == 200, res.json()
+
+            actual = res.content
+            expected = file_path.read_bytes()
+            assert actual == expected
+
+        # If you try to retrieve a file that doesn't exist, we should have a 404 error
+        res = client.get(original_file_url, params={"path": "user/somewhere/something.txt"})
+        assert res.status_code == 404, res.json()
+        assert res.json() == {
+            "description": "'somewhere' not a child of User",
+            "exception": "ChildNotFoundError",
+        }
+
+        # If you try to retrieve a folder, we should get an Error 422
+        res = client.get(original_file_url, params={"path": "user/folder"})
+        assert res.status_code == 422, res.json()
+        assert "Node is a folder node." in res.json()["description"]
+        assert res.json()["exception"] == "PathIsAFolderError"
+
+    def test_retrieve_original_file_from_archive(self, client: TestClient, user_access_token: str) -> None:
```

Comment:
I think it could be better to parametrize this test, archived and unarchived.

Reply:
for the unarchived case it is done in the previous review and I would like to follow the same logic as we had with `raw` API before:
- tests for unarchived study with many file types
- specific test for an archived study